projects/plugins/jetpack/modules/subscriptions/subscribe-modal/class-jetpack-subscribe-modal.php

```diff
@@ -0,0 +1,261 @@
+		$is_fse_theme       = wp_is_block_theme();
+
+		$should_enable_subscriber_modal = ( $is_lettre || $is_newsletter_site ) && $is_fse_theme && $is_modal_enabled;
+
+		/**
+		* Allows force-enabling or disabling the Subscriptions modal.
+		*
+		* @since $$next-version$$
+		*
+		* @param bool $should_enable_subscriber_modal Whether the Subscriptions modal should be enabled.
+		*/
+		return (bool) apply_filters( 'jetpack_subscriptions_modal_force_enabled', $should_enable_subscriber_modal );
+	}
+
+	/**
+	 * Returns true if we we are on frontend of single post.
+	 * Note: Because of how WordPress works, this function will
+	 * only return the correct value after a certain point in the
+	 * WordPress loading process. You cannot, for example, use this
+	 * method in the class contructor method - it will always return false.
+	 *
+	 * @return bool
+	 */
+	public function is_front_end_single_post() {
+		return ! is_admin() && is_single() && 'post' === get_post_type();
+	}
+
+	/**
+	 * Returns the slug for the Subcribe template.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_slug() {
+		return 'subscribe-modal';
+	}
+
+	/**
+	 * Returns the initial content of the Subscribe Modal template.
+	 * This can then be edited by the user.
+	 *
+	 * @return string
+	 */
+	public function get_subscribe_template_content() {
+		return '<!-- wp:group {"style":{"spacing":{"padding":{"top":"50px","bottom":"50px","left":"20px","right":"20px"}}},"layout":{"type":"constrained"}} -->
+		<div class="wp-block-group" style="padding-top:50px;padding-right:20px;padding-bottom:50px;padding-left:20px"><!-- wp:group {"style":{"dimensions":{"minHeight":"0px"},"spacing":{"blockGap":"8px"}},"layout":{"type":"flex","flexWrap":"wrap","justifyContent":"center"}} -->
+		<div class="wp-block-group" style="min-height:0px"><!-- wp:heading {"style":{"typography":{"fontStyle":"normal","fontWeight":"600","fontSize":"26px"},"layout":{"selfStretch":"fit","flexSize":null}}} -->
+		<h2 class="wp-block-heading" style="font-size:26px;font-style:normal;font-weight:600">' . __( 'Disover more from', 'jetpack' ) . '</h2>
+		<!-- /wp:heading -->
+		
+		<!-- wp:site-title {"level":2,"textAlign":"center","style":{"typography":{"fontStyle":"normal","fontWeight":"600","lineHeight":"1.2","fontSize":"26px"}}} /--></div>
+		<!-- /wp:group -->
+		
+		<!-- wp:paragraph {"align":"center","style":{"typography":{"fontSize":"15px"},"spacing":{"margin":{"top":"4px","bottom":"0px"}}}} -->
+		<p class="has-text-align-center" style="margin-top:4px;margin-bottom:0px;font-size:15px">' . __( 'Subscribe to the newsletter to keep reading', 'jetpack' ) . '</p>
+		<!-- /wp:paragraph -->
+		
+		<!-- wp:paragraph {"align":"center","style":{"typography":{"fontSize":"15px"},"spacing":{"margin":{"bottom":"20px","top":"0px"}}}} -->
+		<p class="has-text-align-center" style="margin-top:0px;margin-bottom:20px;font-size:15px">and get access to the full archive.</p>
```

Comment:
We'll probably want to make translatable.

Reply:
Same note about ease of translating https://github.com/Automattic/jetpack/pull/31393#discussion_r1253363857